Code/Tools/FBuild/FBuildCore/Graph/DependencyListNode.cpp

```diff
@@ -0,0 +1,138 @@
+
+// REFLECTION
+//------------------------------------------------------------------------------
+REFLECT_NODE_BEGIN( DependencyListNode, Node, MetaName("Dest") + MetaFile() )
+    REFLECT(        m_Source,                   "Source",                   MetaFile() )
+    REFLECT(        m_Dest,                     "Dest",                     MetaFile() )
+    REFLECT_ARRAY(  m_Patterns,                 "Patterns",                 MetaOptional() )
+    REFLECT_ARRAY(  m_PreBuildDependencyNames,  "PreBuildDependencies",     MetaOptional() + MetaFile() + MetaAllowNonFile() )
+REFLECT_END( DependencyListNode )
+
+// AddDependenciesToList()
+//------------------------------------------------------------------------------
+static void AddDependenciesToList( Array< const AString * > * list , const Array< AString > & patterns , const Dependencies & deps )
+{
+    for ( const Dependency & dep : deps )
+    {
+        if ( dep.IsWeak() == false && dep.GetNode()->IsAFile() )
+        {
+            const AString& filename = dep.GetNode()->GetName();
+
+            for ( const AString & pattern : patterns )
+            {
+                if ( PathUtils::IsWildcardMatch( pattern.Get(), filename.Get() ) )
+                {
+                    list->EmplaceBack( &filename );
+                    break;
+                }
+            }
+        }
+    }
+}
+
+// CONSTRUCTOR
+//------------------------------------------------------------------------------
+DependencyListNode::DependencyListNode()
+: FileNode( AString::GetEmpty(), Node::FLAG_NONE )
+{
+    m_Type = Node::DEPENDENCY_LIST_NODE;
+}
+
+// Initialize
+//------------------------------------------------------------------------------
+/*virtual*/ bool DependencyListNode::Initialize( NodeGraph & nodeGraph, const BFFToken * funcStartIter, const Function * function )
+{
+    // .PreBuildDependencies
+    if ( !InitializePreBuildDependencies( nodeGraph, funcStartIter, function, m_PreBuildDependencyNames ) )
+    {
+        return false; // InitializePreBuildDependencies will have emitted an error
+    }
+
+    // Get node for Source of dependency list
+    if ( !Function::GetFileNode( nodeGraph, funcStartIter, function, m_Source, ".Source", m_StaticDependencies ) )
+    {
+        return false; // GetFileNode will have emitted an error
+    }
+
+    return true;
+}
+
+// DESTRUCTOR
+//------------------------------------------------------------------------------
+DependencyListNode::~DependencyListNode() = default;
+
+// DoBuild
+//------------------------------------------------------------------------------
+/*virtual*/ Node::BuildResult DependencyListNode::DoBuild( Job * /*job*/ )
+{
+    FLOG_VERBOSE( "DEPLIST: '%s' -> '%s'", m_Source.Get(), GetName().Get() );
+
+    // Collect all file dependencies
+    Array< const AString * > dependencyList;
+
+    dependencyList.SetCapacity(
+        GetStaticDependencies().GetSize() +
+        GetDynamicDependencies().GetSize() );
```

Comment:
This node doesn't have any DynamicDependencies (everything is static) so you don't need to try and add stuff from that array.

You should be able to use m_StaticDependencies directly and loop through and write it without copying. You can do the pattern filtering directly in the loop.